Cookbook/Proposals/ManagingAssets.md

```diff
@@ -0,0 +1,58 @@
+
+# Managing Assets
+
+* Author(s): Adrian Śliwa
+* Review Manager: Danilo Aliberti
+
+## Introduction
+
+Babylon's iOS application is a large and advanced software project. This project consists of many frameworks. Each of them consists of many tools or features, most of which need icons or images to be fully implemented. The goal of this proposal is to standardise the way of adding, changing, removing and accessing assets in our codebase.
+
+## Motivation
+
+Managing assets is not difficult engineering problem but without standardised rules we can encounter following issues:
+1. It can be difficult to find if a given icon or image is already included in one of the frameworks. At the moment we have icons and images in many places.
+2. It can cause duplications of icons if the existing icon will be added again.
+3. It could be impossible to reuse icon or image included in one of the frameworks which is not linked to the framework where someone is implementing a feature.
+4. We can end up with many duplicate solutions for accessing icons or images in the code.
+
+To make sure we won't have duplicated icons in the project and all icons will be accessible in all places where we implement the user interface, all of us should agree on uniform rules how and where we should add new assets and how we should manage them.
+
+
+## Proposed solution
+
+The proposed solution contains a set of rules which should be followed during manipulation of our assets:
+1. All new icons and images we are going to use should be added to `BabylonDependencies.framework` in `Assets.xcassets` catalog 
+2. `struct DesignLibrary.Tokens` should have another property for icons `public let icons: Icons`
+3. Then we have to add a new case to `enum ImageIdentifier` which will be embedded in `struct Icons`.
```

Comment:
Shouldn't we just have `struct Icons` + `static let close = Icon(...)` instead of having an `enum ImageIdentifier`? Or make `enum Icons` directly?

Reply:
Having `enum ImageIdentifier` allows us to write the smallest amount of code to add support for a new icon. This thing here is syntactic sugar and I'm open to discussing with the whole team which solution here will be the most convenient one. Today I'm going to check how SwiftGen will fit with @dynamicMemberLookup, maybe it will be the most convenient solution.

Reply:
Do we really need to have this as part of the design library? And why are we keeping this management a manual process? Why not relying on SwiftGen to avoid maintaining all of this ourselves?

Reply:
When I started writing this proposal there was no agreement if we are going to use SwiftGen or not. That's why I decided to include SwiftGen in `Alternatives considered`. Because now we agreed to use SwiftGen I will move it as the proposed solution. We don't have to have it in DesignLibrary, but it's a place where you can access icons now, that's why I kept that solution. Would you like to propose to have it as a separated thing and give some pros of that approach?